repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,131 @@
+# Deploy a high avaliable Repairnator in Kubernetes
+
+![Design](K8sRepairnatorDesign.jpg)
+
+
+## Setup mongodb and ActiveMQ
+First set up mongodb on kubernetes with a persistent memory volume(PVC)(This mean that even if you erase the deployment and redeploy nothing will be lost, the database would look the same)
+
+Then create the mongodb deployemnt using mongodb.yaml in the k8s-mongodb folder .
+
+* kubectl create -f k8s-mongodb/mongodb.yaml
+
+Try it out
+* kubectl get pods 
+Our mongodb pod should be named like "mongo-controller-XXXXX" then we can get into the pod 
+* kubectl exec -it mongo-controller-XXXXX bash
+* mongo
+
+Now deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+* kubectl create -f /queue-for-buildids/activemq.yaml
+To access to the web interface
+* kubectl get pods 
+It should look like activemq-XXXXXXX-XXXXX. Then 
+* kubectl exec -it activemq-XXXXXXX-XXXXX bash
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+* kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+* python publisher.py -d queue-name message-1 message-2 message3 ... message-N
+
+queue-name is according to the format /queue/name. For instance /queue/scanner will send message to the scanner queue.
+You can use this later to manually input build id to the pipeline queue or project to the scanner queue
+
+## Setup scanners and pipelines
+
+To deploy the scanner first go into folder "scanner-dockerimage" then directly build and push the image to docker hub(enviroment variables can be specified or changed later in the "repairnator-scanner.yaml" file. With docker 
+* docker build -t repairnator-scanner:tagname .
```

Comment:
all commands should be in `\`\`\`` blocks. Assume that the reader does not read the text but directly executes the commands one after the other.

Reply:
This issue is addressed in the recent commits.